front/lib/resources/skill/skill_resource.ts

```diff
@@ -1066,24 +1060,28 @@ export class SkillResource extends BaseResource<SkillConfigurationModel> {
       })),
       { transaction }
     );
+
+    // Update the current instance with the new tools to avoid stale data. (Similar to BaseResource update)
+    Object.assign(this, { mcpServerViews });
   }
 
-  async update(
+  private async updateWithAuthorization(
+    auth: Authenticator,
     blob: Partial<Attributes<SkillConfigurationModel>>,
-    transaction?: Transaction
-  ): Promise<[affectedCount: number]> {
-    // TODO(SKILLS 2025-12-12): Refactor BaseResource.update to accept auth.
```

Comment:
I realize I misread it a bit, it was a larger refacto concerning `BaseResource` directly not `SkillResource`.
I guess this is still a good first step wdyt ?